### ISSUE TITLE
Licensing is murky

### ISSUE THREAD

**Issue by Stuk** [developit/proptypes#5]
If this is "literally copied from the React 0.14 codebase" then it's also subject to the patent grant included with that release of React.

It would be good to clarify this.

**Comment by developit** [comment]
Indeed, that's a very good point.  I generally compile PropTypes out, so I might have missed something important for those who don't do the same.  It's possible that PropTypes might be worth a cleanroom implementation of some kind, or perhaps a dummy implementation that simply satisfies the required interface but has no actual functionality would suffice and resolve any license issue.

**Comment by Stuk** [comment]
Yeah, a cleanroom re-implementation would be ideal, but obviously a lot of work.

I think the easiest thing to do would be to add the PATENTS file from the React release this code is from to the repo and add a note to the readme.

The other concern I have is that this as a dependency of preact-compat could cause issues for those using Preact to avoid the patents issue, so it might be worth mentioning that there as well, and the solution of compiling it out.

**Comment by developit** [comment]
That's actually the sole reason `PropTypes` is kept out of the `preact-compat` bundle - licensing. It's just referenced as an external dependency.  It's just exported as a proxy object, but you can alias the import to an empty file/object and entirely disable PropTypes while still using preact-compat.

I'm alright slating a clean implementation here, it's just a bit of a pain point for me since I am not familiar with the API and all the use-cases having avoided it almost entirely myself.  It's the best option though.  I am not even sure we're allowed to copy the PATENTS file - it might constitute asserting ownership, whereas this repo currently is just a mirror.

Another option might be to find a way to replace this "extracted" code with a build job that does the extraction from React's source.  I'm not sure what the effect would be on licensing, but it'd be a build tool as opposed to a copy of the original code.

**Comment by sapegin** [comment]
In any case the PATENTS file is mentioned in the source but not present which feels bad. I also think you can’t just publish this module under the BSD license because you’re using the code which is licensed by a license that requires a patent grant.

Do we still need this package at all? Since Facebook released their own prop-types package and since this one doesn’t really solve licensing issue, I don’t see any benefits of this package anymore.

**Comment by sapegin** [comment]
I think the right way to solve this is to relicense prop-types without the PATENTS file and deprecate this package. This would allow any Preact-like library to use the same package and would solve all licensing issues of this package.

See https://github.com/facebook/react/issues/10191 as well.
